### PR TITLE
Master ctauto294

### PR DESCRIPTION
**Pull request by jrdavisllbean**
Didier and Eder, the code in this branch is for CTAUTO-294. Please review when you have a free moment and let me know if I need to change anything.  In this class I ported the BaseMobile method named waitUntilLoaderImageDisappear(). I found this method helped when waiting for the BillingAddress page. Would it be possible to add to BaseWeb/BasePageObject.

    // TODO -- Possibly add to BasePageObject or BaseWEB -- JRD 9/23/16
    void waitUntilLoaderImageDisappear() {
        By loadingSpinningImage = By.id("ajaxWait");
        wait.until(ExpectedConditions.invisibilityOfElementLocated(loadingSpinningImage));
        this.waitForJQueryToComplete();
        this.waitForJavaScriptToComplete();
    }

### CODE REVIEW

File: src/main/java/com/llbean/automation/web/pages/desktopNTablet/ShippingAddress.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+    public void enterAddress(String address1, String address2, String city, String stateAbbreviation, String zipCode) {
+        logger.info("Enter addr1: " + address1 + ", addr2: " + address2 + " city: " + city + ",");
+        logger.info("state: " + stateAbbreviation + ", zip: " + zipCode + " into the Shipping Address form.");
+
+        // Need to press tab after entering address to close the similar address list.
+        shippingAddress1.sendKeys(address1,Keys.TAB);
+        shippingAddress2.sendKeys(address2,Keys.TAB);
+        shippingCity.sendKeys(city,Keys.TAB);
+        new Select(shippingState).selectByValue(stateAbbreviation.toUpperCase());
```
Comment (didierllbean) [inline src/main/java/com/llbean/automation/web/pages/desktopNTablet/ShippingAddress.java]: Would you create this a global method in BasePageObject to handle any other similar implementations.
Similar to the code in BasePageObject class at line 107, method **selectItem**

File: src/main/java/com/llbean/automation/web/pages/mobile/ShippingAddressMobile.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        clickGotoBillingAddressScreenButton();
+    }
+
+    // The Save and Continue button appears when Shipping Information has been
+    // filled out and the user goes back to the Shipping Info from another checkout page.
+    // The locator for this button is the same as "Continue to Billing Address".
+    void clickGotoBillingAddressScreenButton() {
+        continueToBillingAddress.click();
+        waitUntilLoaderImageDisappear();
+        wait.until(ExpectedConditions.visibilityOfElementLocated(billingAddressPresenceLocator));
```
Comment (didierllbean) [inline src/main/java/com/llbean/automation/web/pages/mobile/ShippingAddressMobile.java]: Please create this as a global wait at the BasePageObject to use in future implementations

**Comment by didierllbean** [comment]
I'm agree with your proposal of adding a waitUntilLoaderImageDisappear in web. I think the best place to put this is in the BaseWeb Class so using the same implementation that is on BaseMobile

**Comment by jrdavisllbean** [comment]
Added selectItem(String textToSelect, WebElement selectField) to BasePageObject, waitUntilLoaderImageDisappear() to BaseWeb, implemented feedback in the Shipping Address classes

File: src/main/java/com/llbean/automation/web/pages/desktopNTablet/BaseWeb.java [hunk 3]
```diff
@@ -18,4 +22,12 @@ public BaseWeb(WebDriver driver) {
     public Header getHeader() {
         return new Header(this.getDriver());
     }
+
+    /*This wait until the loading image disapear from the page*/
+    public void waitUntilLoaderImageDisappear(){
+        WebDriverWait wait = new WebDriverWait(this.getDriver(), 15);
```
Comment (didierllbean) [inline src/main/java/com/llbean/automation/web/pages/desktopNTablet/BaseWeb.java]: Hi @jrdavisllbean  you can remove this line and just use wait. due to there is a variable wait in the BasePageObject, implementing a Fluent wait so this declaration is not needed

**Pull request merged**